tests/core_test.py

```diff
@@ -174,10 +174,13 @@ def foo(x):
     pe.trace_to_jaxpr(foo, (_,))
 
   def test_tree_multimap(self):
-    xs = ({'a': 1}, [2, 3])
-    ys = ({'a': 10}, [20, 30])
-    ys_bad = ({'a': 10, 'b': 10}, [20, 30])
-    zs = ({'a': 11}, [22, 33])
+    class MyDict(dict):
```

Comment:
Instead of replacing the previous test, can you add a new one? It can just copy and paste things from the existing test. The more tests, the better!